### ISSUE TITLE
Lutris becomes unresposive and shows a corrupted window after terminating a game

### ISSUE THREAD

**Issue by lavadrop** [lutris/lutris#2730]
Lutris becomes unresposive at several stages when full debugging is activated AFTER closing my game. When it comes back, activating the "Show logs" option freezes lutris again and then after it comes back, the log window is corrupted. Closing such window agains freezes lutris for a few minutes.

**Expected behavior**

I expect lutris to not freeze and just show the full debug without window corruption.

**Current behavior**
Lutris becomes unresposive and shows a corrupted log window.

**Steps to reproduce**
- Launch lutris from terminal with -d flag
- Configure my game for full debug (since normal debug does not show any info other than pid starting and closing
-Run game until window turns blank
-Force quit game by closing window
-Pressing the "show logs" button

**Lutris debugging output**

https://gist.github.com/lavadrop/e7a96fd3daa983f3f6014c35ad1198dd

**System information**

[lutris-issue-report.zip](https://github.com/lutris/lutris/files/4458313/lutris-issue-report.zip)

**Screenshots**
![Screenshot from 2020-04-09 14-26-34](https://user-images.githubusercontent.com/6760790/78934751-13c82b80-7a71-11ea-97a0-3146a2b6e1ad.png)

**Comment by strycore** [comment]
that's hardcore!
I'm assuming the game produces a very high amount of logs, making the whole program unresponsive. 

* does this also happen when debug logging is disabled in the wine options (I guess not)
* does this happen with other games? I do not have this one and I don't know of a game that could have the same issue. (If the issue indeed is a large amount of logs)

**Comment by tannisroot** [comment]
Personally I haven't seen the corruption but I did encounter the freezes.

**Comment by lavadrop** [comment]
> that's hardcore!
> I'm assuming the game produces a very high amount of logs, making the whole program unresponsive.
> 
>     * does this also happen when debug logging is disabled in the wine options (I guess not)
> 
>     * does this happen with other games? I do not have this one and I don't know of a game that could have the same issue. (If the issue indeed is a large amount of logs)

Indeed, hardcore to the mega!
-No
-Yes. The issue with my game is that it's a very old (2000) game and although it installs fine, there are no errors logged using normal logging.

**Comment by tannisroot** [comment]
@strycore i've found that setting WINEDEBUG envar to +trace easily reproduces the issue.

**Comment by strycore** [comment]
Yeah, using +trace produces an insane amount of logs (and should never be used by anyone except wine devs).
I'll try to repro this locally

**Comment by tannisroot** [comment]
Maybe we should have a limit on the number of lines that should be kept in the output, and truncate the log if it gets over this limit.
This might also solve the issue with Lutris's high ram usage, as I imagine this is where it comes from.
Gnome Terminal limits scrollback to 10K lines, which seems like a good max value.
